File: client/src/app/models/donor.ts
import { Gift } from "./gift";

export class Donor {
    constructor(name: string, detailes?: string, phone?: string, email?: string, logo?: string, showMe: boolean = false) {
        this.name = name;
        this.details = detailes;
        this.phone = phone;
        this.email = email;
        this.logo = logo;
        this.showMe = showMe;
    }

    
    id : number = -1
    name!: string;
    details?: string;
    phone?: string;
    email?: string;
    logo?: string;
    showMe?: boolean;
    gifts?: Gift[];
}

File: server/server/Models/Donor.cs
using System.ComponentModel.DataAnnotations;

namespace server.Models
{
    public class Donor
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Details { get; set; }
        public string? Phone { get; set; }
        public string? Email { get; set; }
        public string? Logo { get; set; }
        public bool? ShowMe { get; set; } = false;

        public List<DonorGift> DonorGifts { get; set; }

    }
}
